src/monitor/formation_metadata.c

```diff
@@ -520,3 +511,116 @@ IsCitusFormation(AutoFailoverFormation *formation)
 {
 	return formation->kind == FORMATION_KIND_CITUS;
 }
+
+/*
+ * set_formation_number_sync_standbys sets number_sync_standbys property of a formation.
+ * The function returns true on success.
+ */
+Datum
+set_formation_number_sync_standbys(PG_FUNCTION_ARGS)
+{
+	text *formationIdText = PG_GETARG_TEXT_P(0);
+	char *formationId = text_to_cstring(formationIdText);
+	int number_sync_standbys = -1;
+	AutoFailoverFormation *formation = 	GetFormation(formationId);
+	bool success = false;
+
+	if (formation == NULL)
+	{
+		ereport(ERROR, (ERRCODE_INVALID_PARAMETER_VALUE,
+						errmsg("unknown formation \"%s\"", formationId)));
+	}
+
+
+	number_sync_standbys = PG_GETARG_INT32(1);
+
+	if (number_sync_standbys < 0)
+	{
+		ereport(ERROR, (ERRCODE_INVALID_PARAMETER_VALUE,
+						errmsg("invalid value for  number_sync_stanbys \"%d\" "
+							   "expected a non-negative integer", number_sync_standbys)));
+	}
+
+	success = SetFormationNumberSyncStandbys(formationId, number_sync_standbys);
+
+	PG_RETURN_BOOL(success);
+}
+
+
+/*
+ * SetFormationNumberSyncStandbys sets numberSyncStandbys property
+ * of a formation entry. Returns true if successfull.
+ */
+bool
+SetFormationNumberSyncStandbys(const char *formationId, int numberSyncStandbys)
+{
+	Oid argTypes[] = {
+			INT4OID, /* numberSyncStanbys */
+			TEXTOID	 /* formationId */
+	};
+
+	Datum argValues[] = {
+			Int32GetDatum(numberSyncStandbys), /* numberSyncStanbys */
+			CStringGetTextDatum(formationId)  /* formationId */
+	};
+	const int argCount = sizeof(argValues) / sizeof(argValues[0]);
+	int spiStatus = 0;
+
+	const char *updateQuery =
+			"UPDATE " AUTO_FAILOVER_FORMATION_TABLE
+			" SET number_sync_standbys = $1"
+			" WHERE formationid = $2";
+
+	SPI_connect();
+
+	spiStatus = SPI_execute_with_args(updateQuery,
+									  argCount, argTypes, argValues,
+									  NULL, false, 0);
+	SPI_finish();
+
+	if (spiStatus != SPI_OK_UPDATE)
+	{
+		elog(ERROR, "could not update " AUTO_FAILOVER_FORMATION_TABLE);
+		return false;
+	}
+
+	return true;
+}
+
+
+/*
+ * AutoFailoverFormationGetDatum prepares a Datum from given formation.
+ * Caller is expected to provide fcinfo structure that contains compatible
+ * call result type.
+ */
+Datum
+AutoFailoverFormationGetDatum(FunctionCallInfo fcinfo, AutoFailoverFormation *formation)
```

Comment:
I didn't say it before, I like the way you restructured that part of the code. Makes it clearer, thanks!